akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorContextSpec.scala

```diff
@@ -1,867 +1,603 @@
+
+      val parentRef = spawn(parent)
+      parentRef ! MakeChild
+      val childRef = probe.expectMessageType[ChildMade].ref
+      childRef ! Fail
+      probe.expectMessage(GotChildSignal(PreRestart))
+      childRef ! Inert
+      probe.expectMessage(InertEvent)
+      childRef ! Ping
+      probe.expectMessage(Pong)
+      parentRef ! StopRef(childRef)
+      probe.expectMessage(GotSignal(Terminated(childRef)(null)))
     }
 
     "stop a child actor" in {
-      sync(setup("ctx04") { (ctx, startWith) ⇒
-        val self = ctx.self
-        startWith.mkChild(Some("A"), ctx.spawnMessageAdapter(ChildEvent), self, inert = true) {
-          case (subj, child) ⇒
-            subj ! Kill(child, self)
-            child
-        }.expectMessageKeep(expectTimeout) { (msg, child) ⇒
-          msg should ===(Killed)
-          ctx.watch(child)
-        }.expectTermination(expectTimeout) { (t, child) ⇒
-          t.ref should ===(child)
-        }
-      })
+      val probe = TestProbe[Event]()
+
+      val child: Behavior[Command] = Behaviors.empty[Command].decorate
+      val parent: Behavior[Command] = (Behaviors.immutablePartial[Command] {
+        case (ctx, MakeChild) ⇒
+          val childRef = ctx.spawnAnonymous(
+            Behaviors.supervise(child).onFailure(SupervisorStrategy.restart)
+          )
+          ctx.watch(childRef)
+          probe.ref ! ChildMade(childRef)
+          Behaviors.same
+        case (ctx, StopRef(ref)) ⇒
+          ctx.stop(ref)
+          Behaviors.same
+      } onSignal {
+        case (_, signal) ⇒
+          probe.ref ! GotSignal(signal)
+          Behavior.stopped
+      }).decorate
+      val parentRef = spawn(parent)
+      parentRef ! MakeChild
+      val childRef = probe.expectMessageType[ChildMade].ref
+      parentRef ! StopRef(childRef)
+      probe.expectMessage(GotSignal(Terminated(childRef)(null)))
     }
 
-    "reset behavior upon Restart" in {
-      sync(setup("ctx05", Some(Behaviors.supervise(_).onFailure(SupervisorStrategy.restart))) { (ctx, startWith) ⇒
-        val self = ctx.self
-        val ex = new Exception("KABOOM05")
-        startWith
-          .stimulate(_ ! BecomeInert(self), _ ⇒ BecameInert)
-          .stimulate(_ ! Ping(self), _ ⇒ Pong2) { subj ⇒
-            muteExpectedException[Exception]("KABOOM05")
-            subj ! Throw(ex)
-            subj
-          }
-          .stimulate(_ ! Ping(self), _ ⇒ Pong1)
-      })
+    "reset behavior upon restart" in {
+      val probe = TestProbe[Event]()
+      val internal = Behaviors.immutablePartial[Command] {
+        case (_, Ping) ⇒
+          probe.ref ! Pong
+          Behavior.same
+        case (_, Fail) ⇒
+          throw new RuntimeException("Boom")
+      }.decorate
+      val behavior = Behaviors.supervise(internal).onFailure(SupervisorStrategy.restart)
+      val actor = spawn(behavior)
+      actor ! Ping
+      probe.expectMessage(Pong)
+      actor ! Fail
+      actor ! Ping
+      probe.expectMessage(Pong)
```

Comment:
This test does not seem to test the same thing anymore, it should test that when the `internal` behavior is restarted it is the initial state, the old one did this by switching to the `Inert` behavior, but I think something simpler could be done, for example having a counter, and responding to `Ping`s with the counter value.